app/controllers/story_image_controller.rb

```diff
@@ -0,0 +1,27 @@
+# typed: false
+
+class StoryImageController < ApplicationController
+  CACHE_DIR = Rails.public_path.join("story_image/").to_s.freeze
+
+  def show
+    short_id = params[:short_id]
+
+    s = Story.where(short_id: short_id).first!
+
+    cached_image = File.join(CACHE_DIR, "#{File.basename(s.short_id)}.png")
+
+    if File.exist?(cached_image)
+      response.headers["Expires"] = 1.hour.from_now.httpdate
+      send_file cached_image, type: "image/png", disposition: "inline"
+    else
+      CreateStoryCardJob.perform_later(s)
```

Comment:
```suggestion
```

I understand the goal here, that you want to fill the cache, but it looks like the job doesn't generate an image if the story doesn't have one (or it's too large, etc).

So then every time someone tries to load the image from us we'd enqueue a job that's going to hit the story URL again, seems like a bad behavior for a bot. We should trust that your PR queues the job every time a story URL is edited, and we can fix any missing paths if a prod bug becomes apparent.